File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_e3/type2_renamed_1_max_subarray.java
public class Solution {
    public int find_highest_subarray(int[] array_list) {
        int absolute_max = array_list[0];
        int running_total = array_list[0];
        
        for (int index = 1; index < array_list.length; index++) {
            running_total = Math.max(array_list[index], running_total + array_list[index]);
            absolute_max = Math.max(absolute_max, running_total);
        }
        return absolute_max;
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_e3/type2_renamed_1_max_subarray.py
def find_highest_subarray(array_list):
    absolute_max = array_list[0]
    running_total = array_list[0]
    
    for index in range(1, len(array_list)):
        running_total = max(array_list[index], running_total + array_list[index])
        absolute_max = max(absolute_max, running_total)
        
    return absolute_max
